### PR TITLE
add timeout parameter

### LINKED ISSUE #3: Add a parameter for setting the timeout duration

### PR DESCRIPTION
**Pull request by conorbros**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -182,6 +190,8 @@ impl DockerCompose {
                     .unwrap();
                 service.log_to_wait_for.is_match(&log)
             }) {
+                let time_to_complete = instant.elapsed();
+                println!("All services ready in {}", time_to_complete.as_secs());
```
Comment (rukai) [inline src/lib.rs]: this should be an info log

**Comment by rukai** [comment]
seems conceptually reasonable, having a hardcoded 120s timeout has been fine for us, but as a generic library we really shouldnt enforce that.
can you resolve the merge conflict before I review it further?

File: src/lib.rs (hunk 1 above)
Comment (conorbros) [reply src/lib.rs]: Thanks

**Review by rukai — APPROVED** [review]
nice!

**Comment by rukai** [comment]
For the 2nd reviewer:
This PR demonstrates the changes that will need to occur in shotover repo after this change https://github.com/shotover/shotover-proxy/pull/1280/files
(Ignoring the mutability change which is from a previous PR)

**Review by benbromhead — APPROVED** [review]

**Pull request merged**